builtin/providers/azurerm/resource_arm_subnet_test.go

```diff
@@ -30,6 +31,44 @@ func TestAccAzureRMSubnet_basic(t *testing.T) {
 	})
 }
 
+func TestAccAzureRMSubnet_update(t *testing.T) {
+
+	ri := acctest.RandInt()
+	preConfig := fmt.Sprintf(testAccAzureRMSubnet_basic, ri, ri, ri, ri, ri)
+	postConfig := fmt.Sprintf(testAccAzureRMSubnet_update, ri, ri, ri, ri, ri)
+	subscriptionID := os.Getenv("ARM_SUBSCRIPTION_ID")
+	nsg_id := fmt.Sprintf("/subscriptions/%s/resourceGroups/acctestRG-%d/providers/Microsoft.Network/networkSecurityGroups/acctestNSG%d", subscriptionID, ri, ri)
+	rt_id := fmt.Sprintf("/subscriptions/%s/resourceGroups/acctestRG-%d/providers/Microsoft.Network/routeTables/acctestRT%d", subscriptionID, ri, ri)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMSubnetDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: preConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMSubnetExists("azurerm_subnet.test"),
+					resource.TestCheckResourceAttr(
+						"azurerm_subnet.test", "network_security_group_id", nsg_id),
+					resource.TestCheckResourceAttr(
+						"azurerm_subnet.test", "route_table_id", rt_id),
+				),
+			},
+			resource.TestStep{
+				Config: postConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMSubnetExists("azurerm_subnet.test"),
+					resource.TestCheckResourceAttr(
+						"azurerm_subnet.test", "network_security_group_id", ""),
```

Comment:
There is a potential dependency issue when going from one NSG to another NSG on the subnet. I am still investigating this, but  I consider it a different bug. Going from `assigned NSG` to `no NSG` is still a valid update.